### PR TITLE
fix(rome_js_analyzer):  precedence for `useOptionalChain`

### LINKED ISSUE #3257: 🐛 invalid code action of `useOptionalChain`
### Environment information

```bash
linux 
AMD
```

### What happened?

```js
(new foo || {}).bar
```
This code snippet will be fixed as 
```
new foo?.bar
```

which is invalid syntax in typescript https://www.typescriptlang.org/play?#code/HYUw7gBAZg9jD8A6ARgQwE4Cgg

### Expected result

Should not be fixed as 
`new foo?.bar`

### Code of Conduct

- [X] I agree to follow Rome's Code of Conduct

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

This PR fixes #3257 

## Test Plan

The PR add two new test cases

### CODE REVIEW

**Review by leops — COMMENTED** [review]
I'm not sure those are really false positive, it looks more like the fix is simply incorrectly removing the parenthesis: `(new foo || {}).bar` should be replaced with `(new foo)?.bar`, but the generated code action emits `new foo?.bar` instead which then gets re-parsed as `new (foo?.bar)` and results in a syntax error

**Comment by ematipico** [comment]
Turns out that the issue was the precedence. If you check the [MDN table](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence#table) the `new` keyword has two different precedence if there or not arguments.

The absence of arguments yields a unique precedence just for this case, which was not tracked in our enum. I updated the code and it now works as @leops suggested.

File: crates/rome_js_syntax/src/lib.rs [hunk 1]
```diff
@@ -293,10 +293,12 @@ pub enum OperatorPrecedence {
     Exponential = 15,
     Unary = 16,
     Update = 17,
-    LeftHandSide = 18,
-    Member = 19,
-    Primary = 20,
-    Group = 21,
+    // `new` without arguments list
+    New = 18,
```
Comment (MichaReiser) [inline crates/rome_js_syntax/src/lib.rs:297]: The name `new` might be misleading. Someone could assume that this is the default precedence for the majority of `new` expressions which is not the case. That's why I would prefer a more explicit name: `NewWithoutArguments`

**Review by MichaReiser — APPROVED** [review]
Would you mind testing if the formatter handles the precedence correctly and, if not, fix the parentheses setting in its own PR?

**Pull request merged**